src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -12,1099 +11,823 @@
-            {
-                "System",
-                "System.Collections.Generic",
-                "System.Globalization",
-                "Microsoft.Extensions.Configuration"
-            };
+        private readonly SourceGenSpec _sourceGenSpec = new();
+        private readonly ImmutableArray<BinderInvocation> _invocations;
+        private readonly HashSet<ITypeSymbol> _unsupportedTypes = new(SymbolEqualityComparer.Default);
+        private readonly Dictionary<ITypeSymbol, TypeSpec?> _createdSpecs = new(SymbolEqualityComparer.Default);
 
-            private BinderMethodSpecifier _methodsToGen;
+        public Parser(SourceProductionContext context, KnownTypeSymbols typeSymbols, ImmutableArray<BinderInvocation> invocations)
+        {
+            Context = context;
+            TypeSymbols = typeSymbols;
+            _invocations = invocations;
+        }
 
-            public Parser(SourceProductionContext context, KnownTypeSymbols typeSymbols)
-            {
-                _context = context;
-                _typeSymbols = typeSymbols;
-            }
+        public SourceProductionContext Context { get; }
+        public KnownTypeSymbols TypeSymbols { get; }
 
-            public SourceGenerationSpec? GetSourceGenerationSpec(ImmutableArray<BinderInvocationOperation> operations)
+        public SourceGenSpec? GetSourceGenerationSpec()
+        {
+            if (TypeSymbols.IConfiguration is null)
             {
-                if (_typeSymbols.IConfiguration is null || _typeSymbols.IServiceCollection is null)
-                {
-                    return null;
-                }
-
-                foreach (BinderInvocationOperation operation in operations)
-                {
-                    IInvocationOperation invocationOperation = operation.InvocationOperation!;
-                    if (!invocationOperation.TargetMethod.IsExtensionMethod)
-                    {
-                        continue;
-                    }
-
-                    switch (operation.MethodGroup)
-                    {
-                        case BinderMethodSpecifier.Bind:
-                            {
-                                ProcessBindCall(operation);
-                            }
-                            break;
-                        case BinderMethodSpecifier.Get:
-                            {
-                                ProcessGetCall(operation);
-                            }
-                            break;
-                        case BinderMethodSpecifier.GetValue:
-                            {
-                                ProcessGetValueCall(operation);
-                            }
-                            break;
-                        case BinderMethodSpecifier.Configure:
-                            {
-                                ProcessConfigureCall(operation);
-                            }
-                            break;
```

Comment:
All this logic supported parsing `ConfigurationBinder` method invocations. It's being moved to a dedicated file under `GeneratedMethods`. Each extension class has it's own file in that directory. Other diffs in this file are cosmetic.

This file now
- coordinates method invocation parsing from each ext. class
- continues the function of creating/registering generation specs for target config types parsed from the invocations